### PR TITLE
chore: Add Beta Deploy and Refactor

### PR DESCRIPTION
**Pull request by akinsola-guardian**
## What does this change?
- Adding a beta-release workflow file
- Refactoring the yml files (creating workflow_call for tests, checks, build and release) to separate processes. 
## Why?
Enable beta releases to improve testing in consumers. 
Separating code to ensure it can be reused in different workflows.

## Further Changes
I'll need to update the required check names as they've been updated. This will be done just before merging to not disrupt other PR's
- Build => CMP CI / Build CMP / Build
- Linting => CMP CI / Run checks / Linting
- Types => CMP CI / Run checks / Types
- Unit tests => CMP CI / Run tests / Unit tests
- e2e tests => CMP CI / Run tests / e2e tests

### CODE REVIEW

File: .github/workflows/README.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
+# Workflow Directory
```
Comment (kenoir) [inline .github/workflows/README.md:1]: This is really neat. Should it be linked to from the root `README.md` to make it more discoverable?

**Review by kenoir — APPROVED** [review]
Looks good to me.

File: .github/workflows/cmp-beta-release-on-label.yml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+jobs:
+  has-permission:
+    name: Check permissions
+    runs-on: ubuntu-latest
+    steps:
+      - name: 'Check if user has admin access'
+        uses: 'lannonbr/repo-permission-check-action@2.0.2'
+        with:
+          permission: 'admin'
```
Comment (sndrs) [inline .github/workflows/cmp-beta-release-on-label.yml:20]: which user is this checking here? does it rely on something having admin rights?

**Review by sndrs — COMMENTED** [review]
this is really interesting, and a nice logical layout. i have a question about permissions which i've attached the relevant line

File: .github/workflows/cmp-beta-release-on-label.yml (hunk 2 above)
Comment (sndrs) [reply .github/workflows/cmp-beta-release-on-label.yml:20]: cc @akash1810

Comment (akash1810) [reply .github/workflows/cmp-beta-release-on-label.yml:20]: Yeah, this makes me nervous, especially as the `GITHUB_TOKEN` we're providing this third-party action with has been scoped with `permissions: write-all`.

**Pull request merged**